### PR TITLE
Domains: Incoming Transfers: Add UI feedback for domain lock status

### LINKED ISSUE #20063: Domains: When re-checking lock status on incoming transfer there's no feedback for the user
Originally reported by @mikeshelton1503 in https://github.com/Automattic/wp-calypso/pull/19949#issuecomment-345363738

> When you click I've unlocked my domain, but the domain hasn't been, there's no feedback from the UI.

lock status check

We should show some kind of notice to the user reminding them to unlock the domain before we can get started.

#### Steps to reproduce
1. Start at `/domains/add/transfer` and select a site.
2. Enter a domain that is currently locked at the registrar.
3. On the pre-check screen, click `I've unlocked my domain`.

#manual-testing

### PR DESCRIPTION
**Pull request by danhauk**
Fixes #20063. There's no feedback for the user in the UI when the domain is still locked and `I've unlocked my domain` is clicked. The button just re-enabled without anything else changing. This made it unclear to the user that anything was actually happening.

Bouncing some ideas around with @klimeryk we came up with this solution. It moves the `Locked` status next to the button to make it more visible, and changes it to `Checking...`. Now when we're checking the lock status the user can see the feedback and clearly know it's been refreshed rather than just broken.

I've also added a line in the description copy to set an expectation that it might take a few minutes for the lock status to update after making changes at the registrar.

lock status check

email refresh

#### Testing

1. Start at `/domains/add/transfer` and select a site.
2. Enter a domain that is currently locked at the registrar.
3. On the pre-check screen, click I've unlocked my domain.
4. Assert the interaction looks like the screenshot above.

### CODE REVIEW

File: client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx [hunk 1]
```diff
@@ -121,24 +124,33 @@ class TransferDomainPrecheck extends React.PureComponent {
 			? translate( 'Your domain is unlocked at your current registrar.' )
 			: translate(
 					"Your domain is locked to prevent unauthorized transfers. You'll need to unlock " +
-						'it at your current domiain provider before we can move it. {{a}}Here are instructions for unlocking it{{/a}}',
+						'it at your current domiain provider before we can move it. {{a}}Here are instructions for unlocking it{{/a}}. ' +
```
Comment (aidvu) [inline client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx]: domiain => domain

Comment (delputnam) [reply client/components/domains/transfer-domain-step/transfer-domain-precheck.jsx]: Ah. Good catch.

**Comment by aidvu** [comment]
Also fixes https://github.com/Automattic/wp-calypso/issues/19979

**Comment by danhauk** [comment]
Refresh email looks good 👍 I might go in later and add some kind of an animation to the email address if it changes just to draw more attention to it. But that's not a blocker for merging this.

**Pull request merged**